plugins/commands/provision/command.rb

```diff
@@ -27,6 +30,20 @@ def execute
         # Go over each VM and provision!
         @logger.debug("'provision' each target VM...")
         with_target_vms(argv) do |machine|
+          if options[:auto_up] && machine.state.id != :running
+            @logger.debug("current machine state `#{machine.state.id}`. Attempting to bring up.")
+            case machine.state.id
+            when :not_created, :poweroff
+              machine.action(:up)
+            when :paused, :saved
+              machine.action(:resume)
+            else
+              raise Vagrant::Errors::ProvisionAutoUpFailure.new(
+                machine_name: machine.name,
+                machine_state: machine.state.short_description
+              )
+            end
```

Comment:
This actually isn't a safe way to do this since providers can return arbitrary state IDs. 

Instead, I would fix this in the provider level with `action_provision` bringing up the machine if necessary.... thoguhts?